webgpu.h

```diff
@@ -0,0 +1,861 @@
+    void const * nextInChain;
+    uint32_t bindGroupLayoutCount;
+    WGPUBindGroupLayout const * bindGroupLayouts;
+} WGPUPipelineLayoutDescriptor;
+
+typedef struct WGPUPipelineStageDescriptor {
+    void const * nextInChain;
+    WGPUShaderModule module;
+    char const * entryPoint;
+} WGPUPipelineStageDescriptor;
+
+typedef struct WGPURasterizationStateDescriptor {
+    void const * nextInChain;
+    WGPUFrontFace frontFace;
+    WGPUCullMode cullMode;
+    int32_t depthBias;
+    float depthBiasSlopeScale;
+    float depthBiasClamp;
+} WGPURasterizationStateDescriptor;
+
+typedef struct WGPURenderPassDepthStencilAttachmentDescriptor {
+    WGPUTextureView attachment;
+    WGPULoadOp depthLoadOp;
+    WGPUStoreOp depthStoreOp;
+    float clearDepth;
+    WGPULoadOp stencilLoadOp;
+    WGPUStoreOp stencilStoreOp;
+    uint32_t clearStencil;
+} WGPURenderPassDepthStencilAttachmentDescriptor;
+
+typedef struct WGPUSamplerDescriptor {
+    void const * nextInChain;
+    WGPUAddressMode addressModeU;
+    WGPUAddressMode addressModeV;
+    WGPUAddressMode addressModeW;
+    WGPUFilterMode magFilter;
+    WGPUFilterMode minFilter;
+    WGPUFilterMode mipmapFilter;
+    float lodMinClamp;
+    float lodMaxClamp;
+    WGPUCompareFunction compare;
+} WGPUSamplerDescriptor;
+
+typedef struct WGPUShaderModuleDescriptor {
+    void const * nextInChain;
+    uint32_t codeSize;
+    uint32_t const * code;
+} WGPUShaderModuleDescriptor;
+
+typedef struct WGPUStencilStateFaceDescriptor {
+    WGPUCompareFunction compare;
+    WGPUStencilOperation failOp;
+    WGPUStencilOperation depthFailOp;
+    WGPUStencilOperation passOp;
+} WGPUStencilStateFaceDescriptor;
+
+typedef struct WGPUTextureViewDescriptor {
+    void const * nextInChain;
+    WGPUTextureFormat format;
+    WGPUTextureViewDimension dimension;
+    uint32_t baseMipLevel;
+    uint32_t mipLevelCount;
+    uint32_t baseArrayLayer;
+    uint32_t arrayLayerCount;
+    WGPUTextureAspect aspect;
+} WGPUTextureViewDescriptor;
+
+typedef struct WGPUVertexAttributeDescriptor {
+    uint32_t shaderLocation;
+    uint64_t offset;
+    WGPUVertexFormat format;
+} WGPUVertexAttributeDescriptor;
+
+typedef struct WGPUBindGroupDescriptor {
+    void const * nextInChain;
+    WGPUBindGroupLayout layout;
+    uint32_t bindingCount;
+    WGPUBindGroupBinding const * bindings;
+} WGPUBindGroupDescriptor;
+
+typedef struct WGPUBindGroupLayoutDescriptor {
+    void const * nextInChain;
+    uint32_t bindingCount;
+    WGPUBindGroupLayoutBinding const * bindings;
+} WGPUBindGroupLayoutDescriptor;
+
+typedef struct WGPUColorStateDescriptor {
+    void const * nextInChain;
+    WGPUTextureFormat format;
+    WGPUBlendDescriptor alphaBlend;
+    WGPUBlendDescriptor colorBlend;
+    WGPUColorWriteMaskFlags writeMask;
+} WGPUColorStateDescriptor;
+
+typedef struct WGPUComputePipelineDescriptor {
+    void const * nextInChain;
+    WGPUPipelineLayout layout;
+    WGPUPipelineStageDescriptor const * computeStage;
```

Comment:
This could probably be `WGPUPipelineStageDescriptor computeStage` to match `vertexStage`

Reply:
Done